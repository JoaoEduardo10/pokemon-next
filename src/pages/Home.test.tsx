import { render } from '@testing-library/react';
import Home from './index';

const mockPokemon = [
	{
		id: 1,
		name: 'picachu',
		url: 'https://pokemon'
	},
	{
		id: 2,
		name: 'picachu',
		url: 'https://pokemon'
	},
];
describe('<Home />', () => {
	it('should render Home', () => {
		const { container } = render(<Home pokemons={mockPokemon} />);

		expect(container).toMatchSnapshot();
	});

	it('should not render Home', () => {
		const { container } = render(<Home pokemons={[]} />);

		expect(container).toMatchSnapshot();
	});
});