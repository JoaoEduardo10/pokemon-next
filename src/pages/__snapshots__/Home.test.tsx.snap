// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should not render Home 1`] = `
<div>
  <div
    class="title_conteiner"
  >
    <h1
      class="title"
    >
      Poke
      <span>
        Next
      </span>
    </h1>
    <img
      alt="pokemon"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fpokeball.png&w=128&q=75"
      srcset="/_next/image?url=%2Fimages%2Fpokeball.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fpokeball.png&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
  </div>
  <ul
    class="pokemon_conteiner"
  />
</div>
`;

exports[`<Home /> should render Home 1`] = `
<div>
  <div
    class="title_conteiner"
  >
    <h1
      class="title"
    >
      Poke
      <span>
        Next
      </span>
    </h1>
    <img
      alt="pokemon"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fpokeball.png&w=128&q=75"
      srcset="/_next/image?url=%2Fimages%2Fpokeball.png&w=64&q=75 1x, /_next/image?url=%2Fimages%2Fpokeball.png&w=128&q=75 2x"
      style="color: transparent;"
      width="50"
    />
  </div>
  <ul
    class="pokemon_conteiner"
  >
    <li
      class="card"
    >
      <img
        alt="picachu"
        data-nimg="1"
        decoding="async"
        height="180"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75 2x"
        style="color: transparent;"
        width="180"
      />
      <p
        aria-label="numero do pokemon"
        class="id"
      >
        #
        1
      </p>
      <h3
        class="title"
      >
        picachu
      </h3>
      <a
        class="btn"
        href="/pokemon/1"
      >
        Detalhes
      </a>
    </li>
    <li
      class="card"
    >
      <img
        alt="picachu"
        data-nimg="1"
        decoding="async"
        height="180"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F2.png&w=384&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F2.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F2.png&w=384&q=75 2x"
        style="color: transparent;"
        width="180"
      />
      <p
        aria-label="numero do pokemon"
        class="id"
      >
        #
        2
      </p>
      <h3
        class="title"
      >
        picachu
      </h3>
      <a
        class="btn"
        href="/pokemon/2"
      >
        Detalhes
      </a>
    </li>
  </ul>
</div>
`;
